samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts

```diff
@@ -202,6 +207,31 @@ export const useChat = () => {
         }
     };
 
+    const filterChats = async (searchString: string) => {
+        const filteredChats: Conversations = {};
+        for (var key in conversations) {
+            if (conversations[key].title.toLowerCase().includes(searchString.toLowerCase())) {
+                filteredChats[key] = conversations[key];
+            }
+        }
+
+        dispatch(
+            setConversations({
+                conversations: conversations,
+                filteredConversations: filteredChats,
+            }),
+        );
+    };
+
+    const clearConversationsFilter = async () => {
+        dispatch(
+            setConversations({
+                conversations: conversations,
+                filteredConversations: conversations,
```

Comment:
Should filteredConversations be a local component state instead of presents in app state? Also I'm curious the difference between displayedConversations vs filteredConversations.

Reply:
https://github.com/microsoft/semantic-kernel/pull/1248#pullrequestreview-1446913040

Reply:
Great recommendation! I completely overengineered this. I moved the filteredCoversations to local component state in ChatList